components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint/src/main/webapp/dynamic_prompt.jsp

```diff
@@ -0,0 +1,136 @@
+    String dataStr = request.getParameter("data");
+    Map data = null;
+    try {
+        data = DynamicPromptUtil.inflateJson(dataStr);
+    } catch (DataFormatException e) {
+        response.sendRedirect("retry.do");
+    }
+    String templatePath = templateMap.get(templateId);
+%>
+
+<html>
+<head>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title><%=AuthenticationEndpointUtil.i18n(resourceBundle, "wso2.identity.server")%>
+    </title>
+    
+    <link rel="icon" href="images/favicon.png" type="image/x-icon"/>
+    <link href="libs/bootstrap_3.3.5/css/bootstrap.min.css" rel="stylesheet">
+    <link href="css/Roboto.css" rel="stylesheet">
+    <link href="css/custom-common.css" rel="stylesheet">
+    
+    <!--[if lt IE 9]>
+    <script src="js/html5shiv.min.js"></script>
+    <script src="js/respond.min.js"></script>
+    <![endif]-->
+</head>
+<body>
+
+
+<!-- header -->
+<header class="header header-default">
+    <div class="container-fluid"><br></div>
+    <div class="container-fluid">
+        <div class="pull-left brand float-remove-xs text-center-xs">
+            <a href="#">
+                <img src="images/logo-inverse.svg" alt="wso2" title="wso2" class="logo">
+                
+                <h1><em><%=AuthenticationEndpointUtil.i18n(resourceBundle, "identity.server")%>
+                </em></h1>
+            </a>
+        </div>
+    </div>
+</header>
+<!-- page content -->
+<div class="container-fluid body-wrapper">
+    
+    <div class="row">
+        <div class="col-md-12">
+    
+            <%
+                if (templatePath != null) {
+            %>
+            <div class="container col-xs-10 col-sm-6 col-md-6 col-lg-4 col-centered wr-content wr-login col-centered">
+                <c:set var="data" value="<%=data%>" scope="request"/>
+                <c:set var="promptId" value="<%=URLEncoder.encode(promptId, StandardCharsets.UTF_8.name())%>" scope="request"/>
+                <jsp:include page="<%=templatePath%>"/>
+            </div>
+    
+            <%
+            } else {
+            %>
+            <div class="container col-xs-7 col-sm-5 col-md-4 col-lg-3 col-centered wr-content wr-login col-centered">
+                <div>
+                    <h2 class="wr-title uppercase blue-bg padding-double white boarder-bottom-blue margin-none"><%=Encode.forHtmlContent("Incorrect Request")%> </h2>
+                </div>
+        
+                <div class="boarder-all col-lg-12 padding-top-double padding-bottom-double error-alert  ">
+                    <div class="font-medium">
+                        <strong>
+                            <%=AuthenticationEndpointUtil.i18n(resourceBundle, "attention")%> :
+                        </strong>
+                    </div>
+                    <div class="padding-bottom-double">
+                        No template found with the given templateId. Please contact the system administrator
```

Comment:
IMO, end user should not see 'template' because he's not aware of such. Shall we change this to have "Page not found, Please contact the system administrator"? We can have this message with template name in the log.